include/seqan3/search/detail/search.hpp

```diff
@@ -180,24 +180,17 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
 template <typename index_t, typename queries_t, typename configuration_t>
 inline auto search_all(index_t const & index, queries_t && queries, configuration_t const & cfg)
 {
-    if constexpr (!std::ranges::forward_range<std::ranges::range_value_t<queries_t>>)
-    {
-        return search_all(index, std::views::single(std::forward<queries_t>(queries)), cfg);
-    }
-    else
-    {
-        // return type: for each query: a vector of text_positions (or cursors)
-        // delegate params: text_position (or cursor). we will withhold all hits of one query anyway to filter
-        //                  duplicates. more efficient to call delegate once with one vector instead of calling
-        //                  delegate for each hit separately at once.
-        using query_t = std::ranges::range_value_t<queries_t>;
-        using single_query_result_t = decltype(search_single(index, std::declval<query_t &>(), cfg));
-        using search_fn_t = std::function<single_query_result_t(query_t &)>;
+    // return type: for each query: a vector of text_positions (or cursors)
+    // delegate params: text_position (or cursor). we will withhold all hits of one query anyway to filter
+    //                  duplicates. more efficient to call delegate once with one vector instead of calling
+    //                  delegate for each hit separately at once.
+    using query_t = std::ranges::range_value_t<queries_t>;
+    using single_query_result_t = decltype(search_single(index, std::declval<query_t &>(), cfg));
+    using search_fn_t = std::function<single_query_result_t(query_t &)>;
 
-        search_fn_t search_fn = [&, cfg] (query_t & query) { return search_single(index, query, cfg); };
+    search_fn_t search_fn = [&, cfg] (query_t & query) { return search_single(index, query, cfg); };
```

Comment:
```suggestion
    std::algorithm search_fn = [&, cfg] (query_t query) { return search_single(index, std::forward<query_t>(query), cfg); };
```

Reply:
From the feeling this shoud be how you would invoke it. So somewhere we iterate over the queries and call `search_single(*queries_it)`. That means this should capture the reference type of the queries range and forward it to the `search_single`. In case this is an l-value reference we forward it as such otherwise we move it and have perfect forwarding. But please look into the search_range implementation to verify that it is called like this.

Reply:
I don't know `std::algorithm`. Can you point me to the documentation?

Reply:
ah that should have been std::function of course